### PR TITLE
Update mono-symbolicate README and man

### PR DESCRIPTION
**Pull request by esdrubal**
Updates mono-symbolicate README and man

Also adds to README.md an example on how to use mono-symbolicate.

### CODE REVIEW

File: mcs/tools/mono-symbolicate/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+to retrieve the symbols mapping code into line numbers from a provided symbol directory.
+
+When `mono-symbolicate` is called with a symbol directory and a file containing a stacktrace:
+``` mono-symbolicate <msym dir> <input file> ```
+The tool writes into stdout the file contents while adding file location to stack frames when
+it is possible to symbolicate with the symbols available on the symbol directory.
+
+## Symbol directory
+The symbol directory contains subfolder named as a MVID or AOTID.
+ - MVID subfolders contain .ddl/.exe and .mdb files.
```
Comment (akoeplinger) [inline mcs/tools/mono-symbolicate/README.md]: ~~.ddl~~ .dll

File: mcs/tools/mono-symbolicate/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+ - AOTID subfolder contain .msym files.
+
+Managed assemblies can be added by calling `mono-symbolicate` with the command `store-symbols`:
+```
+mono-symbolicate store-symbols <msym dir> [<dir>]+
+```
+
+.msym are generated and stored automatically in an AOTID sufolder at the same time the assembly
```
Comment (akoeplinger) [inline mcs/tools/mono-symbolicate/README.md]: ~~sufolder~~ subfolder

File: mcs/tools/mono-symbolicate/README.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+For the pratical example we will use the csharp file `Test/StackTraceDumper.cs` which contains a program that
+outputs a larger number of stacktraces.
+
+The first step is to compile our program with debug data.
+```
+mkdir example
+mcs -debug -out:example/Program.exe Test/StackTraceDumper.cs
+```
+
+Next we need to create the symbol directory and store all the symbols we might need while simbolicating.
```
Comment (akoeplinger) [inline mcs/tools/mono-symbolicate/README.md]: ~~simbolicating~~ symbolicating

File: mcs/tools/mono-symbolicate/README.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+After running `mono-symbolicate store-symbols` command the directory `example/msym-dir` should have a subdirectory
+named as a minified GUID containing the Program.exe and Program.mdb.
+
+We deleted the mdb file so on our next step the generated stack trace won't have line numbers.
+```
+mono example/Program.exe > example/out
+```
+
+Doing `cat example/out` shows use a large number of stacktraces without any line number.
```
Comment (akoeplinger) [inline mcs/tools/mono-symbolicate/README.md]: ~~use~~ us

**Comment by akoeplinger** [comment]
Just a few typos, other than that it LGTM :+1:

File: mcs/tools/mono-symbolicate/README.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+mono-symbolicate store-symbols <msym dir> [<dir>]+
+```
+
+.msym are generated and stored automatically in an AOTID sufolder at the same time the assembly
+is compiled ahead of time, by running a command such as:
+```
+mono --aot=msym-dir=<msym dir>
+```
+
+## Pratical example
```
Comment (evincarofautumn) [inline mcs/tools/mono-symbolicate/README.md]: Practical

File: mcs/tools/mono-symbolicate/README.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+If you do not have `mono-symbolicate` installed on your system you can set it to the built one by doing:
+```
+alias mono="MONO_PATH=../../class/lib/net_4_x ../../../runtime/mono-wrapper"
+alias mono-symbolicate="mono ../../class/lib/net_4_x/mono-symbolicate.exe"
+```
+
+For the pratical example we will use the csharp file `Test/StackTraceDumper.cs` which contains a program that
```
Comment (evincarofautumn) [inline mcs/tools/mono-symbolicate/README.md]: practical

**Comment by esdrubal** [comment]
@alexander and @evincarofautumn thanks for the typo corrections.

**Pull request merged**